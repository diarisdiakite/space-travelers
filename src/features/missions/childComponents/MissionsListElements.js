import React from 'react';
import { useDispatch } from 'react-redux';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import Table from 'react-bootstrap/Table';
import { Button } from 'react-bootstrap';
import Badge from 'react-bootstrap/Badge';
import { joinMission, cancelMissionParticipation } from '../missionsSlice';

function MissionsListElements({ missions }) {
  const dispatch = useDispatch();

  if (!missions) {
    return (
      <div>
        Missions not found
      </div>
    );
  }

  return (
    <Table striped bordered hover>
      <thead>
        <tr>
          <th>Mission</th>
          <th>Description</th>
          <th>Status</th>
          <th>Action</th>
        </tr>
      </thead>
      <tbody>
        {missions.map((mission) => (
          <tr key={mission.mission_id}>
            <td><Link to={`/missions/${mission.mission_id}`} className="no-style bolded">{mission.mission_name}</Link></td>
            <td>{mission.description}</td>
            <td className="fixed-width">
              {!mission.reserved && (
              <Badge bg="secondary">
                NOT A MEMBER
              </Badge>
              )}
              {mission.reserved && (
              <Badge bg="success">
                Active Member
              </Badge>
              )}
            </td>
            <td className="fixed-width">
              {!mission.reserved && (
              <Button
                type="button"
                variant="outline-secondary"
                aria-label="Join mission"
                onClick={() => dispatch(joinMission(mission.mission_id))}
              >
                Join Mission
              </Button>
              )}
              {mission.reserved && (
              <Button
                type="button"
                variant="outline-danger"
                aria-label="cancel mission participation"
                onClick={() => dispatch(cancelMissionParticipation(mission.mission_id))}
              >
                Leave Mission
              </Button>
              )}
            </td>
          </tr>
        ))}
      </tbody>
    </Table>
  );
}

MissionsListElements.propTypes = {
  missions: PropTypes.arrayOf(
    PropTypes.shape({
      mission_id: PropTypes.string.isRequired,
      description: PropTypes.string.isRequired,
      mission_name: PropTypes.string.isRequired,
      website: PropTypes.string,
    }),
  ).isRequired,
};

export default MissionsListElements;
