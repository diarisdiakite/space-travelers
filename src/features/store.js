import { configureStore } from '@reduxjs/toolkit';
import missionsReducer from './missions/missionsSlice';
import usersReducer from './users/usersSlice';
import rocketsSlice from './rockets/rocketsSlice';

const store = configureStore({
  reducer: {
    missions: missionsReducer,
    users: usersReducer,
    rockets: rocketsSlice,
  },
});

export default store;
