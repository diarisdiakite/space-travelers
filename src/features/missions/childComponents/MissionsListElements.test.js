import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router';
import store from '../../store';
import MissionsListElements from './MissionsListElements';

describe('MissionsListElements component', () => {
  it('renders correctly', () => {
    /* const missions = [
      {
        mission_id: 1878900,
        mission_name: 'My mission',
        description: 'My mission description',
        website: 'www.my-mission.org',
      },
    ]; */
    const tree = renderer
      .create(
        <Provider store={store}>
          <MemoryRouter>
            <MissionsListElements /* missions={missions} */ />
          </MemoryRouter>
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
