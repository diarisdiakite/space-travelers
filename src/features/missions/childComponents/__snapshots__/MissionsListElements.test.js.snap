// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MissionsListElements component renders correctly 1`] = `
<div>
  Missions not found
</div>
`;
